src/Workspaces/CSharp/Portable/Extensions/SemanticModelExtensions.cs

```diff
@@ -222,6 +222,10 @@ private static bool CanBindToken(SyntaxToken token)
                 {
                     current = castExpression.Expression;
                 }
+                else if (current is InvocationExpressionSyntax invocationExpression && invocationExpression.Parent is ForEachStatementSyntax) 
+                {
+                    current = invocationExpression.Expression;
```

Comment:
this doesn't seem correct.  i don't see why invocations *in* foreach statements are being called out in particular.  i.e. why is this just not checking if you have an invocation expression only and just handling that case?

Note: i think what you are trying to accomplish is the following:

1. If the user has called a method called 'GetXXX' return 'XXX' as the probable name for the expression.

There are two important aspects of this:

1. It is independent of where the invocation expression is located.  i.e. it would work in while loop, return statement, etc. etc.
2. It is reasonable to infer the name of the variable when seeing the GetXXX pattern (but not for other names that invocation expressions might have).

For example, if i have:

```c#
foreach (var v in EnumerateDirectories()) { }
```

And i introduce a local for that invocation, i should *not* get:

```c#
var enumerateDirectories = EnumerateDirectories();
```

Reply:
Thanks for the contribution!  You're on the right path.  But there are def some things that need to be tweaked first to get the experience right.  I'm happy to help guide you through all of this :)

Reply:
> By the way, I wasn't able to run all the VBTests from the xUnit WPFRunner or commandline. It just hangs indefinitly. Can you point out some directions where I can troubleshoot this?

That i have no clue about.   @Pilchie DO you know what's up here?

Reply:
How should the code look like if you apply the fix on this:

`foreach (var v in EnumerateDirectories()) { }`

Reply:
for ones that return with ToCamelCase, we probably want to introduce new method that convert given string to variable name.

Reply:
we have something like this
http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.ExpressionCodeGenerator.cs,46

probably want to create a method that is shared by all others

Reply:
> How should the code look like if you apply the fix on this?
foreach (var v in EnumerateDirectories()) { }

I am fine with enumerateDirectories. currently we get either something like "v" or "enumerable" as suggested variables, so, I don't think we care much for those cases. either you can leave things as it is or do "enumerableDirectories" and etc.

but, try to get better name when we can like GetXXX cases.